aiohttp_xmlrpc/client.py

```diff
@@ -107,5 +107,5 @@ def method(*args, **kwargs):
 
         return method
 
-    def close(self):
-        return self.client.close()
+    async def close(self):
```

Comment:
It’s the same. Previously method returns coroutine it’s awaitable too.

Reply:
Did not mean to resolve.

This is not true, due to the () the non coroutine function tries to execute to ClientSession’s coroutine and fails. To do what you want it would have to be:

return self.client.close

We could do it this way, but what’s the benefit? I find turning it explicitly into a coroutine here makes it a way more obvious API.

Reply:
> This is not true, due to the () the non coroutine function tries to execute ClientSession’s close coroutine and fails. To do what you want it would have to be:
> 
> return self.client.close

It's wrong, the function which returns coroutine is a coroutine function.

For the Interpreter following code has same effect

```
>>> async def foo():
...     return None
...
>>> def bar():
...     return foo()
...
>>> def baz():
...     return bar()
...
>>> import asyncio
>>> loop = asyncio.get_event_loop()
>>> loop.run_until_complete(asyncio.gather(foo(), bar(), baz()))
[None, None, None]
```

That's happening because any coroutine is an instance of the Coroutine class.

```
>>> type(foo())
<class 'coroutine'>
>>> type(bar())
<class 'coroutine'>
>>> type(baz())
<class 'coroutine'>
```

In this case, function just returning something awaitable. If the client will returns `asyncio.Task()` instead of coroutine this code will incompatible. IMHO: modifying a `client.close()` return value is a bad idea.